src/libcore/pin.rs

```diff
@@ -16,29 +16,30 @@
 //! but doesn't allow moving `T`. The pointer value itself (the `Box`) can still be moved,
 //! but the value behind it cannot.
```

Comment:
I feel like we're dancing around the actual topic here. Should we instead/also give the rules for what you can and cannot get `&mut` to? After all, pinning never actually explicitly prevents moves; all it does is not let you get mutable access to particular things, which could _in turn_ be used to move them. I think giving the full explanation will actually help instead of hurt here, because the reader can map that onto their existing understanding.